### PR TITLE
Add toggle filters to the logging page for Flutter and Dart noise

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Screenshot 2024-01-05 at 11 46 46 AM

Fixes https://github.com/flutter/devtools/issues/7025

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 1]
```diff
@@ -548,40 +578,49 @@ class LoggingController extends DisposableController
...
-              !kindArg.matchesValue(log.kind.toLowerCase())) {
-            return false;
-          }
 
-          if (filter.queryFilter.substrings.isNotEmpty) {
-            for (final substring in filter.queryFilter.substrings) {
-              final matchesKind = log.kind.caseInsensitiveContains(substring);
-              if (matchesKind) return true;
+    bool filterCallback(LogData log) {
+      for (final toggleFilter in filter.toggleFilters) {
```
Comment (elliette) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:583]: Could we set this check to a variable to make it easier to understand what it's checking for?

Something like (might be a better way to write this):
```
final matchesToggledFilter = filter.toggleFilters.any(toggleFilter.enabled.value && !toggleFilter.includeCallback(log));
if (matchesToggledFilter) return false;
```

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 2]
```diff
@@ -548,40 +578,49 @@ class LoggingController extends DisposableController
...
+        if (toggleFilter.enabled.value && !toggleFilter.includeCallback(log)) {
+          return false;
+        }
+      }
+
+      final queryFilter = filter.queryFilter;
+      if (!queryFilter.isEmpty) {
+        final kindArg = queryFilter.filterArguments[kindFilterId];
+        if (kindArg != null &&
```
Comment (elliette) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:592]: Ditto for making this a variable

**Review by elliette — APPROVED** [review]
LGTM with a couple suggestions!

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:595]: Added a TODO. I will address in a follow up PR to clean up the query filter arguments in general.

**Pull request merged**